### ISSUE TITLE
challenges configuring Kerberos without SSL

### ISSUE THREAD

**Issue by AdimUser** [fluent/fluent-plugin-kafka#255]
I am planning to send logs to our cooperate elastic search cluster.

I am using td-agent to forward logs to kafka endpoints which is connected to cooperate elastic search cluster. I have installed the fluent-plugin-kafka plugin with td-agent to send logs to kafka endpoints which is connected to cooperate elastic search cluster

LOGS SENDING PATHS AS BELOW:

`
logs --> td-agent+fluent-plugin-kafka --> kafka cluster --> elasticsearch --> kibana
`

Issue is, kafka endpoints are configured in a way to support SASL AUTHENTICATION without supporting SSL, Since logs are only route with cooperate network, I guess no need of have SSL support.

I am getting bellow error :
```
2019-04-30 17:46:39 +0900 [error]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.3.3/bin/fluentd:8:in `<top (required)>'
  2019-04-30 17:46:39 +0900 [error]: #0 /opt/td-agent/embedded/bin/fluentd:23:in `load'
  2019-04-30 17:46:39 +0900 [error]: #0 /opt/td-agent/embedded/bin/fluentd:23:in `<main>'
2019-04-30 17:46:39 +0900 [error]: #0 unexpected error error_class=ArgumentError error="SASL authentication requires that SSL is configured"
  2019-04-30 17:46:39 +0900 [error]: #0 suppressed same stacktrace
2019-04-30 17:46:39 +0900 [info]: Worker 0 finished unexpectedly with status 1
^C
```

My td-agent configs are :

```
<source>
  @type dummy
  dummy {"hello":"world"}
  tag test
</source>

<match test>
  @type kafka2
  brokers               <broker1>:9002,<broker2>:9002,<<broker3>>:9002
  principal             'appuser@STGKAFKA100.GEAP.XXX.COM'
  keytab                'appuser.keytab'
  client_id             'kafka'
  sasl_over_ssl         false
  get_kafka_client_log  true
  <format>
    @type json
  </format>
  topic_key             'stg_app_elk_1'
  get_kafka_client_log  true

  <buffer topic>
    flush_interval 10s
  </buffer>
</match>
```

**Comment by repeatedly** [comment]
From the implementation, `sasl_over_ssl` parameter is not supported with kerberos: https://github.com/fluent/fluent-plugin-kafka/blob/master/lib/fluent/plugin/out_kafka2.rb#L98

**Comment by AdimUser** [comment]
@repeatedly Thanks for response, Would we able to know, that feature will be available in near future ?

**Comment by AdimUser** [comment]
Hi Everyone,

One of my friends has successfully sent the log messages with SASL_PLAINTEXT protocol. I have not test this but just give a try. 

<match test>
  @type kafka2
  brokers               <broker1>:9002,<broker2>:9002,<<broker3>>:9002
  principal             'appuser@STGKAFKA100.GEAP.XXX.COM'
  keytab                'appuser.keytab'
  client_id             'kafka'
  sasl_over_ssl      false
  ssl_ca_cert 
  get_kafka_client_log  true
  <format>
    @type json
  </format>
  topic_key             'stg_app_elk_1'
  get_kafka_client_log  true

  <buffer topic>
    flush_interval 10s
  </buffer>
</match>


Note : Magic do from the ssl_ca_cert , you can keep it empty or pass empty string

**Comment by xidiandb** [comment]
not work

**Comment by kenhys** [comment]
It seems that it was fixed in v0.9.5.
